src/Microsoft.Health.Fhir.SqlServer.Api/Features/Filters/HttpExceptionFilterAttribute.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Net;
+using EnsureThat;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.SqlServer.Api.Features.Filters
+{
+    internal class HttpExceptionFilterAttribute : ActionFilterAttribute
+    {
+        public override void OnActionExecuted(ActionExecutedContext context)
+        {
+        EnsureArg.IsNotNull(context, nameof(context));
+
+        if (context?.Exception == null)
+        {
+            return;
+        }
+
+        if (!context.ExceptionHandled)
+            {
+                var resultJson = new JObject();
+                if (context.Exception != null)
```

Comment:
This should always be true since the code is already checking that `context.Exception` equals null above and returning.